vault/token_store.go

```diff
@@ -1298,15 +1389,29 @@ func (ts *TokenStore) handleCreateAgainstRole(ctx context.Context, req *logical.
 }
 
 func (ts *TokenStore) lookupByAccessor(ctx context.Context, accessor string, tainted bool) (accessorEntry, error) {
-	saltedID, err := ts.SaltID(ctx, accessor)
+	accessorIDHMAC, err := ts.hmac(ctx, accessor)
+	if err != nil {
+		return accessorEntry{}, err
+	}
+	entry, err := ts.lookupByObfuscatedAccessor(ctx, accessorIDHMAC, tainted)
 	if err != nil {
 		return accessorEntry{}, err
 	}
-	return ts.lookupBySaltedAccessor(ctx, saltedID, tainted)
+
+	if entry.TokenID == "" {
+		// This is only here for backwards compatibility
+		saltedAccessor, err := ts.SaltID(ctx, accessor)
+		if err != nil {
+			return accessorEntry{}, err
+		}
+		entry, err = ts.lookupByObfuscatedAccessor(ctx, saltedAccessor, tainted)
+	}
+
+	return entry, err
 }
 
-func (ts *TokenStore) lookupBySaltedAccessor(ctx context.Context, saltedAccessor string, tainted bool) (accessorEntry, error) {
-	entry, err := ts.view.Get(ctx, accessorPrefix+saltedAccessor)
+func (ts *TokenStore) lookupByObfuscatedAccessor(ctx context.Context, obfuscatedAccessor string, tainted bool) (accessorEntry, error) {
+	entry, err := ts.view.Get(ctx, accessorPrefix+obfuscatedAccessor)
 	var aEntry accessorEntry
```

Comment:
It would read a little more idiomatically if this were a line up.

Reply:
Oh, I just mean that if the accessorEntry line were up one, it wouldn't separate the err from where it's checked. It's not really important.

Reply:
Ah, I see. I don't like it either :-) Fixed!